storage/replica.go

```diff
@@ -186,6 +197,10 @@ type Replica struct {
 		value roachpb.ReplicaDescriptor
 	}
 	truncatedState unsafe.Pointer // *roachpb.RaftTruncatedState
+
+	applyQueueChan chan committedCmd // Command queue
+	stopFlagLock   sync.Mutex        // Protect stopFlag
+	stopFlag       bool              // Mark processReplicaCmd goroutine if already stop
```

Comment:
sorry to come back to this thought, but a `hasProcessReplicaCmdGoroutine int32` updated via `atomic.CompareAndSwapInt32` should avoid having those two fields. I think the resulting code will be clearer.